src/api/auth.ts

```diff
@@ -0,0 +1,12 @@
+import { apiClient } from "./client";
+
+export async function loginApi(
+  email: string,
+  password: string
+): Promise<{ authToken: string; email: string; name: string }> {
+  const response = await apiClient.post("/api/login", { email, password });
+  return response.data.data;
+}
```

Comment:
함수에 직접적으로 return 타입을 명시할수도 있지만, 제네릭을 활용하면 함수의 리턴타입을 자동으로 추론하게끔 할 수 있답니다 ㅎㅎ

참고해주세요!

```suggestion
interface LoginAPIResponse {
  authToken: string;
  email: string;
  name: string;
}

export async function loginApi(
  email: string,
  password: string
) {
  const response = await apiClient.post<{ data: LoginAPIResponse }>("/api/login", {
    email,
    password,
  });
  return response.data.data;
}
```

Reply:
타입스크립트에 대한 지식이 아직 부족해서 AI의 도움을 많이 받았습니다.
피드백 주신 내용과 코드 참고해서 앞으로 더 열심히 공부해 나가겠습니다. 감사합니다!